modules/user_profiles/src/Commands/UserProfilesCommands.php

```diff
@@ -0,0 +1,197 @@
+        $user_blocked_tids = [];
+        foreach ($user_blocked_tid_array as $user_blocked_tid) {
+            $user_blocked_tids[] = $user_blocked_tid['target_id'];
+        }
+        // $this->output()->writeln("  user-to blocked ag tids: " . implode(' ', $user_blocked_tids));
+
+        // get all the affinity groups of $user_from
+        $query = \Drupal::database()->select('flagging', 'fl');
+        $query->condition('fl.uid', $user_from->id());
+        $query->condition('fl.flag_id', 'affinity_group');
+        $query->fields('fl', ['entity_id']);
+        $ag_ids = $query->execute()->fetchCol();
+
+        if ($ag_ids == NULL) {
+            $this->output()->writeln("  from-user is not a member of any affinity groups");
+            return;
+        }
+        // for each affinity group id, add user_to to that affinity group
+        // $this->output()->writeln("  from-user ag ids: " . implode(' ', $ag_ids));
+        foreach ($ag_ids as $ag_id) {
+            $this->addUserToAG($user_to, $ag_id, $user_blocked_tids);
+        }
+    }
+
+    /**
+     * Add a user to an affinity group (unless on the users's block list)
+     */
+    private function addUserToAG(UserInterface $to_user, $ag_id, $user_blocked_tids)
+    {
+        // get the node id of the affinity group
+        $query = \Drupal::database()->select('taxonomy_index', 'ti');
+        $query->condition('ti.tid', $ag_id);
+        $query->fields('ti', ['nid']);
+        $affinity_group_nid = $query->execute()->fetchCol();
+
+        if (!isset($affinity_group_nid[0])) {
+            // not sure how or if this could happen, or what it would mean, but Miles' code in 
+            // CommunityPersonaController.php line 36 also ignores this condition
+            $this->output()->writeln("  *** Warning, from-user flagged as member of affinity group #$ag_id but no such affinity group found - skipping this affinity group");
+            return;
+        }
+
+        // load that affinity group
+        $ag_nid = $affinity_group_nid[0];
+        // $affinity_group_loaded = Node::load($ag_nid);
+        $affinity_group_loaded = \Drupal::entityTypeManager()
+            ->getStorage('node')
+            ->load($ag_nid);
+
+        if (!$affinity_group_loaded) {
+            $this->output()->writeln("  *** Warning, could not load affinity group with node_id #$ag_nid - skipping this affinity group");
+            return;
+        }
+
+        $ag_title = $affinity_group_loaded->getTitle();
+
+        $this->output()->writeln("  From-user member of AG '$ag_title' (id #$ag_id)");
+
+        // get AG taxonomy id
+        $ag_taxonomy = \Drupal::entityTypeManager()
+            ->getStorage('taxonomy_term')
+            ->loadByProperties(['name' => $ag_title]);
+        $ag_taxonomy = reset($ag_taxonomy);
+
+        if (!$ag_taxonomy) {
+            $this->output()->writeln("    *** Warning, no taxonomy id found for AG title '$ag_title'");
+            return;
+        }
```

Comment:
We already have the taxonomy id from the initial flagging. I think we can remove the above code?